bgpd/bgp_filter.c

```diff
@@ -488,6 +491,12 @@ DEFUN (no_ip_as_path,
 	argv_find(argv, argc, "LINE", &idx);
 	regstr = argv_concat(argv, argc, idx);
 
+	if (!config_bgp_aspath_validate(regstr)) {
+		vty_out(vty, "Invalid character in as-path access-list %s\n",
+			regstr);
+		return CMD_WARNING_CONFIG_FAILED;
+	}
+
 	regex = bgp_regcomp(regstr);
```

Comment:
(Not related to your PR: why are we even compiling the regex in the `no ...` command? Just ditch half the code here maybe...)

Also if we remove this, we can put the code from `config_bgp_aspath_validate` straight into `ip_as_path()` since we don't need it here.

Reply:
all fixed, waiting for others about this one